src/coreclr/vm/nativeimage.cpp

```diff
@@ -286,7 +286,7 @@ void NativeImage::CheckAssemblyMvid(Assembly *assembly) const
 
     SString message;
     message.Printf(W("MVID mismatch between loaded assembly '%s' (MVID = %s) and an assembly with the same simple name embedded in the native image '%s' (MVID = %s)"),
-        assembly->GetSimpleName(),
+        SString(SString::Utf8, assembly->GetSimpleName()).GetUnicode(),
```

Comment:
When is the destructor of the `SString` object that holds the string going to be called? 

It has be called after `Printf` finishes. Otherwise, the pointer returned from `GetUnicode` is going to point to freed memory. I do not know whether C++ standard guarantees it or not.

Reply:
I believe that the C++ standard does guarantee that all such destructions happen at the end of the "statement", in this particular case the expression statement calling the <code>Printf</code> function. I'll try to look up the relevant reference in the C++ standard. Alternatively, I can easily make these two local variables if you think it's safer or if it's more in line with typical CoreCLR runtime coding style.

Reply:
While the official C++ standard is not freely available, at least the cppreference web page states in the section

https://en.cppreference.com/w/cpp/language/implicit_conversion#Temporary_materialization

that:

*All temporary objects are destroyed as the last step in evaluating the full-expression that (lexically) contains the point where they were created, and if multiple temporary objects were created, they are destroyed in the order opposite to the order of creation. This is true even if that evaluation ends in throwing an exception.

There are two exceptions from that:

The lifetime of a temporary object may be extended by binding to a const lvalue reference or to an rvalue reference (since C++11), see reference initialization for details.
The lifetime of a temporary object created when evaluating the default arguments of a default or copy constructor used to initialize an element of an array ends before the next element of the array begins initialization.*

Reply:
Sounds good. Thanks for checking.